### PR TITLE
Update CPI overlays for dualstack support

### PR DESCRIPTION
**Pull request by HanFa**
- configures vsphere-cloud-controller-manager with dualstack ip family
  settings
- enables dualstack feature flag in vsphere-cloud-controller-manager

Signed-off-by: Tyler Schultz <tschultz@vmware.com>
Co-authored-by: Aidan Obley <aobley@vmware.com>
Co-authored-by: Fang Han <fhan@vmware.com>

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by christianang — APPROVED** [review]
LGTM

File: pkg/v1/providers/ytt/02_addons/cpi/cpi_addon_data.lib.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ vsphereCPI:
   #! For now, use the first ipFamily when TKG_IP_FAMILY is a dual stack value (ipv4,ipv6 or ipv6,ipv4)
   #! TODO: stop doing this when this issue is resolved:
   #! https://github.com/kubernetes/cloud-provider-vsphere/issues/302
-  ipFamily: #@ data.values.TKG_IP_FAMILY.split(",")[0]
+  ipFamily: #@ data.values.TKG_IP_FAMILY
```
Comment (flawedmatrix) [inline pkg/v1/providers/ytt/02_addons/cpi/cpi_addon_data.lib.yaml:22]: Should we remove the comments preceding this line?

**Review by flawedmatrix — COMMENTED** [review]
Other than my one comment, looks good to me.

**Comment by mcwumbly** [comment]
We are calling this `do-not-merge/hold` for now because:

1. We need to update the vendir'd version of the vsphere-cpi to 1.22.3 first. We plan to do this in a separate PR.
2. We want to further validate that the `ENABLE_ALPHA_DUAL_STACK` feature flag gets persisted on the vsphere-cpi after kapp-controller takes over and reinstalls it.

**Review by tylerschultz — APPROVED** [review]

**Comment by christianang** [comment]
Some of the work that has been done in this PR is redundant now that the package itself sets the dualstack feature flag. Pulling in the new package is being done as part of #1199. @tylerschultz and I can work together to pull in the changes from here that are missing: notably removing the ipFamily split.

@HanFa we can just close this PR in favor of getting all the changes in #1199

**Comment by HanFa** [comment]
close this PR as it's superseded by https://github.com/vmware-tanzu/tanzu-framework/pull/1199

**Pull request closed without merging**